### PR TITLE
[cli] Rewrite Unit tests to TypeScript + Jest

### PR DESCRIPTION
**Pull request by TooTallNate**
* Rewrites the CLI unit tests to be TypeScript and use Jest (consistent with the unit tests in the other packages in this repo).
* The file structure of the new test files mirrors that of the `src` directory, so that it's easy to find the relevant tests.
* Code coverage is also properly set up through Jest - you can already see a big increase in coverage from this PR.
* Adds a mock API server framework, with the intention of making it possible to write granular tests of the CLI commands. Using mocks also has the benefit of not requiring `VERCEL_TOKEN` env var to be set, so 3rd party PRs will be able to have their tests run. Ultimately this will also help with test coverage, since we will be writing unit tests that test the commands' code directly.
* Converts `Output` into a proper class (which is subclassed for the mocks).

### CODE REVIEW

File: packages/cli/test/mocks/client.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      apiUrl: '',
+      authConfig: {},
+      output: new Output(),
+      config: {},
+      localConfig: {},
+    });
+    this.mockOutput = jest.fn();
+
+    this.app = express();
+    this.app.use(express.json());
```
Comment (styfle) [inline packages/cli/test/mocks/client.ts:33]: Why use `express` here? We don't really use it anywhere else as far as I know?

File: packages/cli/src/util/get-user.ts [hunk 2]
```diff
@@ -2,11 +2,7 @@ import Client from './client';
 import { User } from '../types';
 import { APIError, InvalidToken, MissingUser } from './errors-ts';
 
-let user: User | undefined;
-
 export default async function getUser(client: Client) {
-  if (user) return user;
```
Comment (styfle) [inline packages/cli/src/util/get-user.ts:8]: This used to be cached?

File: packages/cli/src/util/output/create-output.ts [hunk 3]
```diff
@@ -68,112 +77,99 @@ function _createOutput({ debug: debugEnabled = false }: OutputOptions = {}) {
...
-    dim,
-    time,
-    note,
-    spinner: setSpinner,
-    stopSpinner,
   };
 }
+
+export default function createOutput(opts?: OutputOptions) {
+  return new Output(opts);
```
Comment (styfle) [inline packages/cli/src/util/output/create-output.ts:174]: I noticed this is no longer a singleton.

What happens when `createOutput()` is called more than once?

File: packages/cli/test/mocks/client.ts (hunk 1 above)
Comment (TooTallNate) [reply packages/cli/test/mocks/client.ts:33]: Mainly just for the decent router capabilities. Open to suggestions if you'd prefer a different framework.

File: packages/cli/src/util/get-user.ts (hunk 2 above)
Comment (TooTallNate) [reply packages/cli/src/util/get-user.ts:8]: It did, but for not really any reason. This function is invoked exactly once for most commands (and if it's > 1 then the command should be updated to not do that). Also the caching was getting in the way of the unit tests getting unexpectedly cached results 😛

File: packages/cli/src/util/output/create-output.ts (hunk 3 above)
Comment (TooTallNate) [reply packages/cli/src/util/output/create-output.ts:174]: The codebase has been cleaned up at this point to not create many output instances, just a single one gets created in the entrypoint file. The singleton requirement was more of a temporary band-aid while cleaning that part up. I think it's ok to remove that requirement at this point.

**Review by styfle — APPROVED** [review]

**Review by AndyBitz — APPROVED** [review]

**Pull request merged**